molgenis-data-security/src/main/java/org/molgenis/data/security/acl/EntityAclManagerImpl.java

```diff
@@ -253,26 +236,39 @@ private EntityAcl toEntityAcl(Acl acl)
 
 	private EntityAce toEntry(AccessControlEntry accessControlEntry)
 	{
-		Permission permission = toPermission(accessControlEntry.getPermission());
+		Set<Permission> permissions = toPermission(accessControlEntry.getPermission());
 		SecurityId securityId = toSecurityId(accessControlEntry.getSid());
-		return EntityAce.create(permission, securityId, accessControlEntry.isGranting());
+		return EntityAce.create(permissions, securityId, accessControlEntry.isGranting());
 	}
 
 	private SecurityId toSecurityId(Sid sid)
 	{
 		if (sid instanceof PrincipalSid)
 		{
 			String principal = ((PrincipalSid) sid).getPrincipal();
-			return SecurityId.create(principal, null);
+			return SecurityId.createForUsername(principal);
 		}
 		if (sid instanceof GrantedAuthoritySid)
 		{
 			String grantedAuthority = ((GrantedAuthoritySid) sid).getGrantedAuthority();
-			return SecurityId.create(null, grantedAuthority);
+			return SecurityId.createForAuthority(grantedAuthority);
 		}
 		throw new IllegalArgumentException(String.format("Unknown Sid type '%s'", sid.getClass().getSimpleName()));
 	}
 
+	private org.springframework.security.acls.model.Permission toDomainPermission(Set<Permission> permissions)
+	{
+		CumulativePermission result = new CumulativePermission();
+		for (Permission permission : permissions)
+		{
+			if (permission != null) // GSON serializes unknown enum values to null
```

Comment:
@fdlk @tommydeboer Right place for this to happen? EntityAclManagerAcl doesn't care about JSON serialization